### PR TITLE
Kotlin DSL release notes for 5.3

### PR DESCRIPTION
**Pull request by eskatos**

### CODE REVIEW

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -43,6 +43,60 @@ dependencies {
...
+```
+
+It makes use of type-safe accessors for the `java {}` extension, the `checkstyleMain` task and the `testImplementation` configuration, all contributed by the plugins it applies (`java-library` and `checkstyle`).
+
+This plugin can then be applied to regular projects:
+
+```kotlin
+// build.gradle.kts
+plugins {
+    `my-java-convetion`
```
Comment (big-guy) [inline subprojects/docs/src/docs/release/notes.md]: `my-java-convention`?

Comment (eskatos) [reply subprojects/docs/src/docs/release/notes.md]: Good catch, thanks. Fixed

File: subprojects/docs/src/docs/release/notes.md [hunk 2]
```diff
@@ -43,6 +43,60 @@ dependencies {
...
+    checkstyle
+}
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_11
+    targetCompatibility = JavaVersion.VERSION_11
+}
+
+tasks {
+    withType<JavaCompile> {
```
Comment (bamboo) [inline subprojects/docs/src/docs/release/notes.md]: :x: `javaCompile { ...` should work as well

Comment (eskatos) [reply subprojects/docs/src/docs/release/notes.md]: It works but doesn't do the same, I'll leave it as is for the sake of the example.
And it's demonstrated using `checkstyleMain` below.

**Pull request merged**